dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp

```diff
@@ -157,80 +153,95 @@ namespace DYN {
 
   void
   ModelLoadRestorativeWithLimits::getSize() {
-    sizeF_ = 4;
-    sizeY_ = 6;
+    sizeF_ = 3;
+    sizeY_ = 5;
     sizeZ_ = numZ;
     sizeG_ = 3;
     sizeMode_ = 2;
     calculatedVars_.assign(nbCalculatedVariables_, 0);
   }
 
+  void
+  ModelLoadRestorativeWithLimits::evalDynamicYType() {
+    if (UMaxPuReached_ || UMinPuReached_) {
+      yType_[0] = ALGEBRAIC;  // uf
+    } else {
+      yType_[0] = DIFFERENTIAL;  // uf
+    }
+  }
   void
   ModelLoadRestorativeWithLimits::evalStaticYType() {
-    yType_[0] = DIFFERENTIAL;  // differential equation on ufRaw
-    yType_[1] = ALGEBRAIC;  // uf
-    yType_[2] = EXTERNAL;  // ur
-    yType_[3] = EXTERNAL;  // ui
-    yType_[4] = ALGEBRAIC;  // ir
-    yType_[5] = ALGEBRAIC;  // ii
+    yType_[1] = EXTERNAL;  // ur
```

Comment:
Please use the index defined in ::init here

Reply:
the initialization of indexes is now done at constructor, as init is called after evalStaticYType